components/sup/src/http_gateway.rs

```diff
@@ -113,135 +112,193 @@ impl fmt::Display for ListenAddr {
     }
 }
 
-struct ManagerFs;
+#[derive(Default, Serialize)]
+struct HealthCheckBody {
+    status: String,
+    stdout: String,
+    stderr: String,
+}
 
-impl typemap::Key for ManagerFs {
-    type Value = manager::FsCfg;
+impl Into<StatusCode> for HealthCheck {
+    fn into(self) -> StatusCode {
+        match self {
+            HealthCheck::Ok | HealthCheck::Warning => StatusCode::OK,
+            HealthCheck::Critical => StatusCode::SERVICE_UNAVAILABLE,
+            HealthCheck::Unknown => StatusCode::INTERNAL_SERVER_ERROR,
+        }
+    }
 }
 
-pub struct Server(Iron<Chain>, ListenAddr);
+struct AppState {
+    fs_cfg: Arc<manager::FsCfg>,
+    gateway_state: Arc<RwLock<manager::GatewayState>>,
+}
 
-impl Server {
-    pub fn new(manager_state: Arc<manager::FsCfg>, listen_addr: ListenAddr) -> Self {
-        let mut r = Router::new();
-        if !feat::is_enabled(feat::RedactHTTP) {
-            r.get("/butterfly", butterfly, "butterfly");
-            r.get("/census", census, "census");
+impl AppState {
+    fn new(fs_cfg: Arc<manager::FsCfg>, gs: Arc<RwLock<manager::GatewayState>>) -> Self {
+        AppState {
+            fs_cfg: fs_cfg,
+            gateway_state: gs,
         }
+    }
+}
 
-        r.get("/", doc, "doc");
-        r.get("/services", services, "services");
-        r.get("/services/:svc/:group", service, "services_svc_group");
-        r.get(
-            "/services/:svc/:group/:org",
-            service,
-            "services_svc_group_org",
-        );
-        r.get(
-            "/services/:svc/:group/config",
-            config,
-            "services_svc_group_config",
-        );
-        r.get(
-            "/services/:svc/:group/health",
-            health,
-            "services_svc_group_health",
-        );
-        r.get(
-            "/services/:svc/:group/:org/config",
-            config,
-            "services_svc_group_org_config",
-        );
-        r.get(
-            "/services/:svc/:group/:org/health",
-            health,
-            "services_svc_group_org_health",
-        );
+pub struct Server;
 
-        let mut chain = Chain::new(r);
-        chain.link(persistent::Read::<ManagerFs>::both(manager_state));
-        Server(Iron::new(chain), listen_addr)
+impl Server {
+    pub fn run(
+        listen_addr: ListenAddr,
+        fs_cfg: Arc<manager::FsCfg>,
+        gateway_state: Arc<RwLock<manager::GatewayState>>,
+    ) {
+        thread::spawn(move || {
+            let sys = actix::System::new("sup-http-gateway");
+
+            server::new(move || {
+                let app_state = AppState::new(fs_cfg.clone(), gateway_state.clone());
+                App::with_state(app_state).configure(routes)
+            }).bind(listen_addr.to_string())
+            .unwrap()
+            .start();
+
+            let _ = sys.run();
```

Comment:
Because I copied it that way from their docs https://actix.rs/docs/server/ 😁 

`run()` returns an `i32`, so my guess is if you remove the `let _ =` you'll get a warning about an unhandled return value, but I could be wrong.  I'll try it out.

Reply:
Looks fine, no warning.  I'll remove it.